### ISSUE TITLE
Docs suggestion: Outline typical workflow

### ISSUE THREAD

**Issue by liamappelbe** [ingydotnet/git-subrepo#535]
I was having similar issues others have reported, where every time I switch computers and pull down the new changes, I would get spurious merge conflicts.

I realized recently that it was because I was doing things in the wrong order. I have some bash wrapper functions for both pull and push, and in both of them I was doing the outer repo, then all the subrepos.

I've switched the order so that for pushing I do subrepos first then the outer repo, and for pulling I do the outer repo first then the subrepos. In both cases, I do a `git subrepo clean --all` afterwards. Also, while I always pull from every subrepo, I only push to the subrepos I own.

With this new workflow I haven't had any issues. So it might be worth outlining some of this stuff, and any other workflow tips you folks have, in the documentation.

**Comment by andry81** [comment]
Yes, maintaining the pull/push dependencies between repositories, maintaining common user config (user mail, name, commit properties and so on) bitween user computers is a headache in the git (opposed to the centralized scm where it's automatically relyed on the server side). But all of these is not a problem of the git-subrepo, because it inherits all these issues by design of the git. Welcome to the distributed git hell.
